### PR TITLE
Add MySQL DDLStatement antlr visitor

### LINKED ISSUE #3914: Optimization for Sharing Parser with ANTLR Visitor for MySQL
Hi folks, we would like to refactor existing parser engine with **ANTLR visitor** for better performance. It is scheduled as a long-term task and a big change for our 5.x release, **we would like to welcome anyone who is interested in it to join us for this wonderful sharding parser**! 😃 

> Take a look at the task detail below, you would learn it more. ⬇️ 

**Please follow up with your comment 📝 , if you want to join**!  What’s more, welcome your any comment and suggestion!

### PR DESCRIPTION
**Pull request by SteNicholas**
Fixes #3914.

`MySQLVisitor` lacks of visitor for `DDLStatement.g4`. It is necessary to add `DDLStatement.g4` antlr visitor.

Changes proposed in this pull request:
- `MySQLVisitor` adds `DDLStatement.g4` antlr visitor.

### CODE REVIEW

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java [hunk 1]
```diff
@@ -190,6 +227,135 @@ public ASTNode visitCreateUser(final CreateUserContext ctx) {
...
+        return result;
+    }
+
+    @Override
+    public ASTNode visitCreateIndex(final CreateIndexContext ctx) {
+        CreateIndexStatement result = new CreateIndexStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        IndexSegment index = (IndexSegment) visit(ctx.indexName());
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: `META-INF/parsing-rule-definition/mysql/sql-statement-rule-definition.xml` tells us there is no need to add IndexSegment to CreateIndexStatement when DB is MySQL.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java [hunk 2]
```diff
@@ -190,6 +227,135 @@ public ASTNode visitCreateUser(final CreateUserContext ctx) {
...
+        return result;
+    }
+
+    @Override
+    public ASTNode visitDropIndex(final DropIndexContext ctx) {
+        DropIndexStatement result = new DropIndexStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        IndexSegment index = (IndexSegment) visit(ctx.indexName());
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: META-INF/parsing-rule-definition/mysql/sql-statement-rule-definition.xml tells us there is no need to add IndexSegment to CreateIndexStatement when DB is MySQL.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java [hunk 3]
```diff
@@ -190,6 +227,135 @@ public ASTNode visitCreateUser(final CreateUserContext ctx) {
...
 
     // DDLStatement.g4
+    @Override
+    public ASTNode visitCreateTable(final CreateTableContext ctx) {
+        CreateTableStatement result = new CreateTableStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        CreateDefinitionClause_Context createDefinitionClause = ctx.createDefinitionClause_();
+        if (null != createDefinitionClause) {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: IMO, it is better to put it to visitCreateDefinitionClause(). Parent visior has no reposibility to visit its child visitor, and just gets the result returned by its child.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java [hunk 4]
```diff
@@ -190,6 +227,135 @@ public ASTNode visitCreateUser(final CreateUserContext ctx) {
...
+        return result;
+    }
+
+    @Override
+    public ASTNode visitAlterTable(final AlterTableContext ctx) {
+        AlterTableStatement result = new AlterTableStatement();
+        TableSegment table = (TableSegment) visit(ctx.tableName());
+        result.setTable(table);
+        result.getAllSQLSegments().add(table);
+        if (null != ctx.alterDefinitionClause_()) {
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: The above suggestion applied to the following visitors. Could you review those rather long visitors for optimizing?

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java (hunk 1 above)
Comment (SteNicholas) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: @tristaZero Okay, I would like to remove `IndexSegment`.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java (hunk 2 above)
Comment (SteNicholas) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: @tristaZero Okay, I would like to remove `IndexSegment`.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java (hunk 3 above)
Comment (SteNicholas) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: @tristaZero No, there is no any statement match for visitCreateDefinitionClause() result.

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java (hunk 4 above)
Comment (SteNicholas) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: @tristaZero I knew your meaning. I have already previously optimized for the visitor, but no any statement contains `TableSegment`, cause that no match for visit**().

File: shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java [hunk 5]
```diff
@@ -871,6 +1051,53 @@ private ASTNode createExpressionSegment(final ASTNode astNode, final ParserRuleC
...
+            }
+        }
+        for (GeneratedDataType_Context generatedDataType: columnDefinition.generatedDataType_()) {
+            CommonDataTypeOption_Context commonDataTypeOption = generatedDataType.commonDataTypeOption_();
+            if (null != commonDataTypeOption) {
+                if (null != commonDataTypeOption.primaryKey()) {
+                    isPrimaryKey = true;
+                }
+                if (null != commonDataTypeOption.referenceDefinition_()) {
+                    statement.getAllSQLSegments().add((TableSegment) visit(commonDataTypeOption.referenceDefinition_().tableName()));
```
Comment (tristaZero) [inline shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: To create `visitReferenceDefinition_()` is better. Short and simple function always make peple easy to read. Other than, one function is supposed to focus on itself, not include its child.

Comment (tristaZero) [reply shardingsphere-sql-parser/shardingsphere-sql-parser-mysql/src/main/java/org/apache/shardingsphere/sql/parser/MySQLVisitor.java]: Could give more review on other long funcions?

**Pull request merged**